test/functional/legacy/delete_spec.lua

```diff
@@ -98,4 +102,50 @@ describe('Test for delete()', function()
     eq(0, eval("delete('Xdir4/Xfile')"))
     eq(0, eval("delete('Xdir4', 'd')"))
   end)
+
+  it('complicated name delete', function()
+    source([[
+      call mkdir('Xcomplicated')
+      call mkdir('Xcomplicated/[complicated-1 ]')
+      call mkdir('Xcomplicated/{complicated,2 }')
+      split Xcomplicated/Xfile
+      call setline(1, ['a', 'b'])
+      w
+      w Xcomplicated/\[complicated-1\ \]/Xfile
```

Comment:
appveyor error:

    Vim(write):E212: Can't open file for writing (Xcomplicated\\[complicated-1 \]\Xfile)

it would be easier/cleaner to use `writefile()`, and will avoid `:write` escaping problems on Windows/unix:

    call writefile(['Xcomplicated-1', 'content'], 'Xcomplicated/[complicated-1 ]')
    call writefile(['Xcomplicated-2', 'content'], 'Xcomplicated/{complicated,2 }')